### ISSUE TITLE
Issue: Submitting an object with an ICollection to a service results in 415 error (Only with WebAssembly)

### ISSUE THREAD

**Issue by devedse** [servicetitan/Stl.Fusion#186]
Hi all,

Today I've been working on building an application based on the Blazor sample. To make everything a bit easier for me, rather then passing individual values I instead pass around full objects from the Frontend to the Backend.

For example in the `Shop.razor` file:
```
await ShopService.UpdateShop(locals.Current);
```

locals.Current is of type `DbShop`:
```
    public record DbShop : LongKeyedEntity
    {
        [Required, MaxLength(120)]
        public string Name { get; set; } = "";

        public ICollection<DbShopItem> ShopItems { get; set; } = new List<ShopItem>();

        [MaxLength(9999)]
        public string XPathPrice { get; set; } = "";
        [MaxLength(9999)]
        public string XPathAvailability { get; set; } = "";
        [MaxLength(9999)]
        public string AvailabilityText { get; set; } = "";
    }
```

When I execute this code with server side blazor everything runs fine, however if I run this with WebAssembly I see the following error:
```
{type: "https://tools.ietf.org/html/rfc7231#section-6.5.13", title: "Unsupported Media Type",…}
status: 415
title: "Unsupported Media Type"
traceId: "00-71d8e4fd07aae244b05d21e0e0d6c6b5-2b78ddb4e971ba46-00"
type: "https://tools.ietf.org/html/rfc7231#section-6.5.13"
```

This is the request url:
```
http://localhost:5005/api/shop/UpdateShop?shop=DbShop+{+Id+%3D+1%2C+Name+%3D+Alternatef%2C+ShopItems+%3D+System.Collections.Generic.List`1%5BSamples.Blazor.Common.Models.DbShopItem%5D%2C+XPathPrice+%3D+dinggggawefwe%2C+XPathAvailability+%3D+dinggggg%2C+AvailabilityText+%3D++}
```

What I do find strange is the fact that everything is posted in the URL rather then the body. Is this intended?

**Comment by alexyakunin** [comment]
Hi, it works exactly as you "prescribe" in RestEase client interface plus controller. Here is how you switch one of parameters to body payload:

![image](https://user-images.githubusercontent.com/1673423/100093663-b371d880-2e0c-11eb-9f2f-7302c2024a4f.png)
![image](https://user-images.githubusercontent.com/1673423/100093731-cb495c80-2e0c-11eb-8f3d-c0f82c9ea2e1.png)

Besides that, `ICollection<T>` should work as `[ComputeMethod]` parameter by default, but I am not sure if ASP.NET Core will bind such a parameter properly. 

I highly recommend you to switch to simply an array or `List<T>` there - IMO using an interface like `IEnumerable<T>` is actually a bad practice. The reason is: every user of such a type expects a precise behavior of such a collection - i.e. no lazy loading on enumeration, etc., but when you specify a generic interface there, you sort of imply that API is open for such implementations. Even though in reality it doesn't - and this is true in almost any place like this.

In short, avoid using `IEnumerable<T>` and other similar interfaces in POCO & model objects. Just use exact types.

**Comment by alexyakunin** [comment]
@devedse hopefully this helps :)

**Comment by devedse** [comment]
Hey @alexyakunin , I added [Body] to the clients and this solved my problem. Thanks for the tips around the IEnumerable for now I just want to make things work as easy as possible but I'll look into this later.

**Comment by alexyakunin** [comment]
On [FromBody] - maybe it works this way for some param types - I'd rather be explicit about that though...

Anyway, great it's resolved 👍
